core/src/main/java/com/linecorp/armeria/client/limit/ConcurrencyLimit.java

```diff
@@ -73,4 +119,25 @@ static ConcurrencyLimitBuilder builder(int maxConcurrency) {
      * }</pre>
      */
     CompletableFuture<SafeCloseable> acquire(ClientRequestContext ctx);
+
+    class SettableLimit implements Supplier<Integer> {
```

Comment:
Let's extract outside of this interface and rename to `SettableConcurrentLimit`.